lib/python/qmk/cli/cformat.py

```diff
@@ -19,27 +17,46 @@ def cformat(cli):
         if which(binary):
             clang_format[0] = binary
             break
-
-    # Find the list of files to format
-    if cli.args.files:
-        cli.args.files = [os.path.join(os.environ['ORIG_CWD'], file) for file in cli.args.files]
-    else:
-        ignores = ['tmk_core/protocol/usb_hid', 'quantum/template']
-        for dir in ['drivers', 'quantum', 'tests', 'tmk_core']:
-            for dirpath, dirnames, filenames in os.walk(dir):
-                if any(i in dirpath for i in ignores):
-                    dirnames.clear()
-                    continue
-
-                for name in filenames:
-                    if name.endswith(('.c', '.h', '.cpp')):
-                        cli.args.files.append(os.path.join(dirpath, name))
-
-    # Run clang-format on the files we've found
     try:
-        subprocess.run(clang_format + cli.args.files, check=True)
+        if not files:
+            cli.log.warn('No changes detected. Use "qmk cformat -a" to format all files')
+            return False
+        if cli.args.files and cli.args.all_files:
+            cli.log.warning('Filenames passed with -a, only formatting: %s', ','.join(cli.args.files))
+        subprocess.run(clang_format + files, check=True)
         cli.log.info('Successfully formatted the C code.')
 
     except subprocess.CalledProcessError:
         cli.log.error('Error formatting C code!')
         return False
+
+
+@cli.argument('-a', '--all-files', arg_only=True, action='store_true', help='Format all core files.')
+@cli.argument('-b', '--base-branch', default='origin/master', help='Branch to compare to diffs to.')
+@cli.argument('files', nargs='*', arg_only=True, help='Filename(s) to format.')
+@cli.subcommand("Format C code according to QMK's style.")
+def cformat(cli):
+    """Format C code according to QMK's style.
+    """
+    # Empty array for files
+    files = []
+    # Core directories for formatting
+    core_dirs = ['drivers', 'quantum', 'tests', 'tmk_core']
+    ignores = ['tmk_core/protocol/usb_hid', 'quantum/template']
+    # Find the list of files to format
+    if cli.args.files:
+        files.extend(qmk.path.normpath(file) for file in cli.args.files)
+    # If -a is specified
+    elif cli.args.all_files:
+        all_files = qmk.path.cpathfiles(core_dirs)
+        files.extend(file for file in all_files if not any(i in str(file) for i in ignores))
```

Comment:
Wouldn't this be sufficient here? 

```suggestion
        files.extend(file for file in all_files if str(file) not in ignores)
```

Reply:
```
>>> ignores = ['tmk_core/protocol/usb_hid', 'quantum/template']
>>> file = 'quantum/template/test.c'
>>> file not in ignores
True
>>> not any(i in str(file) for i in ignores)
False
```
Logic is slightly different between the two, one doesn't do the substring check properly.

Reply:
The longer I think about, the more your code makes sense.  
Maybe you could comment it, so next time I look at it, it will make sense faster. :)